src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -197,7 +201,10 @@ private void CheckAbstractClassImplementations(BindingDiagnosticBag diagnostics)
                     if (synthesizedImplementation.MethodImpl is { } methodImpl)
                     {
                         Debug.Assert(methodImpl is { Body: not null, Implemented: not null });
-                        methodImpls.Add(methodImpl);
+                        if (implementedMethods.Add(methodImpl.Implemented))
```

Comment:
>if (implementedMethods.Add(methodImpl.Implemented)) [](http://example.com/codeflow?start=24&length=51)

While perhaps this change addresses problem for the specific scenario. The place in combination with the nature of the fix doesn't feel right to me.
Here are the reasons:
- The ```SynthesizedExplicitImplementations``` result produced by this method is used for things other than emit. For example, it looks like it is used by implementation of an ```FindImplementationForInterfaceMember``` API. Therefore, changing result of this method is likely to affect result of ```FindImplementationForInterfaceMember``` as well and the change might have a negative impact on it. I think that the tests accompanying the fix should include tests for all affected consumers. It is quite possible that those consumers do not work in similar scenarios with the current result. Then we need to make sure that the fix also addresses that, assuming the root cause is the same. For example, I would expect that ```FindImplementationForInterfaceMember``` API finds implementation for members from either duplicate interface. This, for example, might not work if we simply drop the information here.
- It looks like we possibly have a similar issue around creating forwarding methods. I.e. instead of emitting duplicate method impls on the same method, we might generate multiple forwarding methods pointing to the same interface member in its method impl. It doesn't look like the change could possibly address that situation.
- If in the end we decide to continue with dropping information in result produced by this method, it might be better to check that we already processed matching interfaceMember and skip it much earlier. For example, why call ```SynthesizeInterfaceMemberImplementation``` if we already know that we are going to drop result it is going to produce? 

It is quite possible, however, that each consumer of the result requires different handling for the duplicates. Then the fix(s) should be moved to the code specific for each consumer. For emit layer, for example, we will filter out duplicates in such a way so that the method impls used match the interface included into ```GetInterfacesToEmit``` set.

Reply:
- I will move the fixes to the consumers.
- I wanted to add tests but I couldn't find how `FindImplementationForInterfaceMember` uses `SynthesizedExplicitImplementations`.
- Forwarding was indeed broken, thanks. I will add a fix in this PR.